### DISCUSSION TITLE
Symfony 2.7 : validating nested entity

### DISCUSSION THREAD

**Discussion by mlwacomos** [symfony/symfony#49784] [Q&A]
Hi there.

I have a class A. One of the property (listB) is an array of B objects (from class B)

In AType class (managing the builder) I add this : 

    $builder->add('listB', 'collection', array(
       'type' => new BType($liste_options),
       'label' => false,
       'options' => array('label' => false, 'attr' => array(
           'class' => 'hascotctpStyle'
       )),
       'required'=> false
   ));

In Atype class, I add "cascade_validation" option to the Resolver, so my embedded forms can be validated. If I dont do that, there is no validation.

The validator is a custom validator applied with annotation on a field of Class B.

Inside the validator class I have this : 

    public function validate($value, Constraint $constraint) {
        // value = value of the Class B field I am validating
        $entity = $this->context->getRoot()->getData();
        ....
    }

I naturally thought that this entity would be the Class B object I am validating because the validator is working on the field of this object. But no, the entity is the parent object Class A.

I dont know how to get the Class B object I am validating. Can you help ?

Thank you.

**Comment by MatTheCat** [comment]
Hi,

you commented

> value = value of the Class B field I am validating

Isn’t that what you want?
